rotkehlchen/globaldb/handler.py

```diff
@@ -1011,6 +1011,127 @@ def add_historical_prices(entries: List['HistoricalPrice']) -> None:
 
         connection.commit()
 
+    @staticmethod
+    def add_single_historical_price(entry: HistoricalPrice) -> bool:
+        """
+        Adds the given historical price entries in the DB.
+        Returns True if the operation succeeded and False otherwise
+        """
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        try:
+            serialized = entry.serialize_for_db()
+            cursor.execute(
+                """INSERT OR IGNORE INTO price_history(
+                from_asset, to_asset, source_type, timestamp, price
+                ) VALUES (?, ?, ?, ?, ?)
+                """,
+                serialized,
+            )
+        except sqlite3.IntegrityError as e:
+            connection.rollback()
+            log.error(
+                f'Failed to add single historical price. {str(e)}. ',
+            )
+            return False
+        # success
+        connection.commit()
+        return True
+
+    @staticmethod
+    def get_manual_prices(
+        from_asset: Optional[Asset],
+        to_asset: Optional[Asset],
+    ) -> List[Dict[str, str]]:
+        """Returns prices added to the database by the user for an asset"""
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        querystr = (
+            'SELECT from_asset, to_asset, source_type, timestamp, price FROM price_history '
+            'WHERE source_type=? '
+        )
+        # Start by adding the manual source type to the list of params
+        params = [HistoricalPriceOracle.MANUAL.serialize_for_db()]  # pylint: disable=no-member
+        if from_asset is not None:
+            querystr += 'AND from_asset=? '
+            params.append(from_asset.identifier)
+        if to_asset is not None:
+            querystr += 'AND to_asset=? '
+            params.append(to_asset.identifier)
+        querystr += 'ORDER BY timestamp'
+        query = cursor.execute(querystr, tuple(params))
+        return [
+            {
+                'from_asset': entry[0],
+                'to_asset': entry[1],
+                'timestamp': entry[3],
+                'price': entry[4],
+            }
+            for entry in query.fetchall()
```

Comment:
hehe you still did not understand what I said. Here you are still running the iteration 2 times. Once in `fetchall()` and once in the loop comprehension.
```suggestion
            for entry in query
```